sim/network.go

```diff
@@ -107,31 +111,94 @@ func (n *Network) SetAlarm(sender f3.ActorID, payload string, at float64) {
 
 func (n *Network) Sign(sender f3.ActorID, msg []byte) []byte {
 	// Fake implementation.
-	// Just prepends 8-byte sender ID to message.
-	buf := new(bytes.Buffer)
-	if err := binary.Write(buf, binary.BigEndian, sender); err != nil {
-		panic(err)
-	}
-	return append(buf.Bytes(), msg...)
+	// Just prepends the pubkey associated with the sender ID to message.
+	aux := append([]byte(nil), n.actor2PubKey[sender]...)
+	return append(aux, msg...)
 }
 
-func (n *Network) Verify(sender f3.ActorID, msg, sig []byte) bool {
+func (n *Network) Verify(pubKey f3.PubKey, msg, sig []byte) bool {
 	// Fake implementation.
-	// Just checks that first 8 bytes of signature match sender ID,
+	// Just checks that first bytes of the signature match sender ID,
 	// and remaining bytes match message.
-	buf := bytes.NewReader(sig)
-	var recoveredSender uint64
-	if err := binary.Read(buf, binary.BigEndian, &recoveredSender); err != nil {
-		return false
+	aux := append([]byte(nil), pubKey...)
+	aux = append(aux, msg...)
+	return bytes.Equal(aux, sig)
+}
+
+func (n *Network) Aggregate(sigs [][]byte, aggSignature []byte) []byte {
+	// Fake implementation.
+	// Just appends signature to aggregate signature.
+	// This fake aggregation is not commutative (order matters)
+	// But determinism is preserved by sorting by weight both here and in VerifyAggregate
```

Comment:
IMO: the deterministic sorting should happen outside the interface to match real signature schemes which can also have this behaviour.

Reply:
But that would mean forcing outside the interface a sorting that may not be required by *all* signature schemes. Would it not be better to have it inside the signature schemes that need it only?

Reply:
(I get optimizing if the real BLS is not commutative, and coupling in that particular instantiation of the interface, but I also mean in particular in the case of the sorting required by this general-purpose (or so-intending) fake aggregation and verification scheme, that just happens to need to sort)

Reply:
I see your point, and I don't know which approach is better at this moment. I mentioned it because this requirement was unclear to me, and I immediately highlighted it as a possible difference between test/sim and reality.

We can keep order independence internal, but then we should highlight it in the interface documentation.

Reply:
> We can keep order independence internal, but then we should highlight it in the interface documentation.

Sounds good, I suggest going back to this when we discuss interfaces in #42 not to duplicate discussions.